pandas/tests/io/parser/test_usecols.py

```diff
@@ -21,6 +21,22 @@
                                "expected but not found: {0}")
 
 
+@pytest.mark.parametrize("names,usecols,missing", [
+    (None, [0, 3], r"\[3\]"),
+    (["a", "b", "c"], [0, -1, 2], r"\[-1\]"),
+    (None, [3], r"\[3\]"),
+    (["a"], [3], r"\[3\]")
+])
+def test_usecols_out_of_bounds(all_parsers, names, usecols, missing):
+    # See gh-25623
+    data = "a,b,c\n1,2,3\n4,5,6"
+    parser = all_parsers
+
+    mssg = _msg_validate_usecols_names.format(missing)
```

Comment:
I'm not tied to it but I find the error reporting and naming here somewhat confusing as it doesn't really distinguish position from labels. Is there a logical way to make that distinction here and obviously in the implementation?

Reply:
I agree, but the purpose of the commit is to provide error reporting consistent with the error raised when the size of _usecols_ is different from the width of the dataframe. 

Distinguishing between reporting mechanisms for labels and indices seems like a new commit.